sourced/ml/repo2/base.py

```diff
@@ -4,6 +4,7 @@
 
 from pyspark import StorageLevel
 from pyspark.sql.types import Row
+from pyspark.sql.functions import col, udf
```

Comment:
`col` is a very widespread name. I suggest to `from pyspark.sql import functions`